ext/mysqli/mysqli.stub.php

```diff
@@ -0,0 +1,773 @@
+<?php
+
+class mysqli_driver
+{
+}
+
+class mysqli
+{
+    /**
+     * @param ?string $host
+     * @param ?string $user
+     * @param ?string $password
+     * @param ?string $database
+     * @param ?int $port
+     * @param ?string $socket
+     */
+    public function __construct(?string $host = null, ?string $user = null, ?string $password = null, ?string $database = null, ?int $port = null, ?string $socket = null);
+
+    /**
+     * @param ?bool $mode
+     *
+     * @return bool
+     */
+    public function autocommit(?bool $mode);
+
+    /**
+     * @param ?int $flags
+     * @param ?string $name
+     *
+     * @return bool
+     */
+	public function begin_transaction(?int $flags = 0, ?string $name = '');
+
+	/**
+	 * @param ?string $user
+	 * @param ?string $password
+	 * @param ?string $database
+	 *
+	 * @return bool
+	 */
+	public function change_user(?string $user, ?string $password, ?string $database);
+
+	/**
+	 * @return string
+	 */
+	public function character_set_name();
+
+    /**
+     * @return bool
+     */
+	public function close();
+
+	/**
+	 * @param ?int $flags
+	 * @param ?string $name
+	 *
+	 * @return bool
+	 */
+	public function commit(?int $flags = 0, ?string $name = '');
+
+    /**
+     * @param ?string $host
+     * @param ?string $user
+     * @param ?string $password
+     * @param ?string $database
+     * @param ?int $port
+     * @param ?string $socket
+     *
+     * @return mysqli
+     */
+	public function connect(?string $host = null, ?string $user = null, ?string $password = null, ?string $database = null, ?int $port = null, ?string $socket = null);
+
+    /**
+     * @return bool
+     */
+	public function dump_debug_info();
+
+	/**
+	 * @param ?string $debug_options
+	 * @return bool
+	 */
+	public function debug(?string $debug_options);
+
+    /**
+     * @return object
+     */
+	public function get_charset();
+
+	/**
+	 * @return string
+	 */
+	public function get_client_info();
+
+	/**
+	 * @return array
+	 */
+	public function get_connection_stats();
+
+	/**
+	 * @return string
+	 */
+	public function get_server_info();
+
+	/**
+	 * @return mysqli_warning
+	 */
+	public function get_warnings();
+
+	/**
+	 * @return mysqli
+	 */
+	public function init();
+
+	/**
+	 * @param ?int $connection_id
+	 *
+	 * @return bool
+	 */
+	public function kill(?int $connection_id);
```

Comment:
The param is not nullable.

Reply:
the test says otherwise https://github.com/php/php-src/blob/master/ext/mysqli/tests/mysqli_class_mysqli_reflection.phpt#L540

Reply:
It's only because currently, the parameter type  is not set (which can be seen from the arginfo):
`ZEND_ARG_INFO(0, connection_id)`.

`ReflectionParameter::allowsNull()` returns true if this is the case: https://github.com/php/php-src/blob/796d11c942ca056b9fb63c90fc200e788a6618ef/ext/reflection/php_reflection.c#L2656

You only have to rely on the ZPP to find it out if a parameter is nullable (whether there is a `!` or not)